### ISSUE TITLE
NEST fails to assert presence of required Boost libraries when Boost is configured

### ISSUE THREAD

**Issue by Helveg** [nest/nest-simulator#3124]
**Describe the bug**
Building NEST with Boost 1.83.0 fails.

**To Reproduce**
1. Install Boost 1.83.0 downloaded from their website
2. Set `BOOST_ROOT` to unpacked location
3. Remove `boost/iterator/iterator_adaptor.hpp`
4. `cmake ..`
5. `make install -j8`

**Desktop/Environment (please complete the following information):**
 - OS: [e.g. iOS, Ubuntu-18.04]: WSL2 
 ```
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.6 LTS
Release:        20.04
Codename:       focal
```
 - Shell: [e.g. bash, zsh]: bash
 - Python-Version: [e.g. Python 3.6] 3.9
 - NEST-Version: master @ 355e52d121ff644c6af568d8df9c07aa7b2b3bcf

**Additional context**

This is the build log

```
(b4py9) robin@LAPTOP-JF6T3PSU:/mnt/c/Users/pwd06/git/nest-simulator/build$ cmake .. -Dwith-mpi=ON
-- The CXX compiler identification is GNU 10.5.0
-- The C compiler identification is GNU 10.5.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for include file inttypes.h
-- Looking for include file inttypes.h - found
-- Looking for include file mach-o/dyld.h
-- Looking for include file mach-o/dyld.h - not found
-- Looking for include file mach/mach.h
-- Looking for include file mach/mach.h - not found
-- Looking for include file memory.h
-- Looking for include file memory.h - found
-- Looking for include file stdint.h
-- Looking for include file stdint.h - found
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for C++ include istream
-- Looking for C++ include istream - found
-- Looking for C++ include ostream
-- Looking for C++ include ostream - found
-- Looking for C++ include sstream
-- Looking for C++ include sstream - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of long long
-- Check size of long long - done
-- Check size of u_int16_t
-- Check size of u_int16_t - done
-- Check size of uint16_t
-- Check size of uint16_t - done
-- Check size of u_int64_t
-- Check size of u_int64_t - done
-- Check size of uint64_t
-- Check size of uint64_t - done
-- Looking for NAN
-- Looking for NAN - found
-- Looking for isnan
-- Looking for isnan - found
-- Looking for M_E
-- Looking for M_E - found
-- Looking for M_PI
-- Looking for M_PI - found
-- Looking for expm1
-- Looking for expm1 - not found
-- Info: Host triple: x86_64-pc-linux 
-- Info: Target triple: x86_64-pc-linux 
-- Found Python: /home/robin/.pyenv/versions/b4py9/bin/python3.9 (found suitable version "3.9.9", minimum required is "3.8") found components: Interpreter Development.Module 
-- Found Cython: /home/robin/.pyenv/versions/b4py9/bin/cython (found suitable version "0.29.33", minimum required is "0.28.3") 
-- Found LTDL: /usr/lib/x86_64-linux-gnu/libltdl.so (found version "2.4.6") 
-- Found Readline: /usr/lib/x86_64-linux-gnu/libreadline.so (found version "8.0")
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Found GSL: /usr/include (found version "2.5") 
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")
-- Found MPI_C: /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so (found version "3.1") 
-- Found MPI_CXX: /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so (found version "3.1") 
-- Found MPI: TRUE (found version "3.1")
-- Found PY_mpi4py: /home/robin/.pyenv/versions/b4py9/lib/python3.9/site-packages/mpi4py  
-- Found Boost: /usr/lib/cmake/Boost-1.83.0/BoostConfig.cmake (found suitable version "1.83.0", minimum required is "1.69.0")  
-- Info: Check the abort exitcode. 
-- Info: Check the abort exitcode. 134 
-- Info: Check the segmentation fault exitcode. 
-- Info: Check the segmentation fault exitcode. 139 
-- Info: Check whether the compiler ignores cmath makros. 
-- Info: Check whether the compiler ignores cmath makros. OFF 
-- Info: Check whether the compiler does NOT include <*.h> headers ISO conformant. 
-- Info: Check whether the compiler does NOT include <*.h> headers ISO conformant. OFF 
-- Info: Check whether the compiler respects symbolic signal names in signal.h. 
-- Info: Check whether the compiler respects symbolic signal names in signal.h. OFF 
-- Info: Check static template member declaration. 
-- Info: Check static template member declaration. OFF 
-- Info: Check for STL vector capacity base unity. 
-- Info: Check for STL vector capacity base unity. ON 
-- Info: Check for STL vector capacity doubling strategy. 
-- Info: Check for STL vector capacity doubling strategy. ON 
-- Info: Check whether the compiler fails with ICE. 
-- Info: Check whether the compiler fails with ICE. OFF 
-- Info: Check if ::nan is available from cmath. 
-- Info: Check if ::nan is available from cmath. ON 
-- Info: Check if ::isnan is available from cmath. 
-- Info: Check if ::isnan is available from cmath. ON 
-- Info: Check if Random123 generators work. 
-- Info: Check if Random123 generators work. ON 
-- Info: Done configuring NEST version: 3.6.0-post0.dev0 

--------------------------------------------------------------------------------
NEST Configuration Summary
--------------------------------------------------------------------------------

Target System       : Linux
Cross Compiling     : FALSE
C compiler          : GNU 10.5.0 (/usr/bin/cc)
C compiler flags    :  -Wall -fopenmp   -pthread -O2 -fdiagnostics-color=auto
C++ compiler        : GNU 10.5.0 (/usr/bin/c++)
C++ compiler flags  :  -std=c++17 -Wall -fopenmp -pthread -O2 -fdiagnostics-color=auto
Build dynamic       : ON

Built-in modelset   : full

User modules        : None

Python bindings     : Yes (Python 3.9.9: /home/robin/.pyenv/versions/b4py9/bin/python3.9)
    Includes        : /home/robin/.pyenv/versions/3.9.9/include/python3.9
    Libraries       :
    Cython          : Yes (Cython 0.29.33: /home/robin/.pyenv/versions/b4py9/bin/cython)
    MPI4Py          : Yes (/home/robin/.pyenv/versions/b4py9/lib/python3.9/site-packages/mpi4py/include)

Documentation       : No

Use threading       : Yes (OpenMP: -fopenmp)
    Libraries       : /usr/lib/gcc/x86_64-linux-gnu/10/libgomp.so;/usr/lib/x86_64-linux-gnu/libpthread.so

Use GSL             : Yes (GSL 2.5)
    Includes        : /usr/include
    Libraries       : /usr/lib/x86_64-linux-gnu/libgsl.so;/usr/lib/x86_64-linux-gnu/libgslcblas.so

Use Readline        : Yes (GNU Readline 8.0)
    Includes        : /usr/include
    Libraries       : /usr/lib/x86_64-linux-gnu/libreadline.so;/usr/lib/x86_64-linux-gnu/libncurses.so

Use libltdl         : Yes (LTDL 2.4.6)
    Includes        : /usr/include
    Libraries       : /usr/lib/x86_64-linux-gnu/libltdl.so

Use MPI             : Yes (MPI: /usr/bin/mpicxx)
    Includes        : /usr/lib/x86_64-linux-gnu/openmpi/include/openmpi;/usr/lib/x86_64-linux-gnu/openmpi/include
    Libraries       : /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so;/usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so
    Compile Flags   : -pthread
    Link Flags      : -pthread
    Launcher        : /usr/bin/mpiexec -n <np> <prog> <args>

Detailed timers     : No

Use MUSIC           : No

Use libneurosim     : No

Use Boost           : Yes (Boost 1.83.0)
    Includes        : /usr/include
    Libraries       :

Use SIONlib         : No

Use HDF5            : No

For details on setting specific flags for your MPI launcher command, see the
CMake documentation at https://cmake.org/cmake/help/latest/module/FindMPI.html

--------------------------------------------------------------------------------

The NEST executable will be installed to:
  /home/robin/.pyenv/versions/b4py9/bin/

NEST dynamic libraries and user modules will be installed to:
  /home/robin/.pyenv/versions/b4py9/lib/nest/

PyNEST will be installed to:
    /home/robin/.pyenv/versions/b4py9/lib/python3.9/site-packages

To set necessary environment variables, add the following line
to your ~/.bashrc :
  source /home/robin/.pyenv/versions/b4py9/bin/nest_vars.sh

--------------------------------------------------------------------------------

You can now build and install NEST with
  make
  make install
  make installcheck

If you experience problems with the installation or the use of NEST,
please see https://www.nest-simulator.org/frequently_asked_questions
or go to https://www.nest-simulator.org/community to find out how to
join the user mailing list.

-- Configuring done (28.7s)
-- Generating done (2.1s)
-- Build files have been written to: /mnt/c/Users/pwd06/git/nest-simulator/build
(b4py9) robin@LAPTOP-JF6T3PSU:/mnt/c/Users/pwd06/git/nest-simulator/build$ make install -j8
[  0%] Building CXX object libnestutil/CMakeFiles/nestutil.dir/iaf_propagator.cpp.o
[  0%] Building CXX object libnestutil/CMakeFiles/nestutil.dir/logging_event.cpp.o
[  1%] Building CXX object libnestutil/CMakeFiles/nestutil.dir/numerics.cpp.o
[  1%] Building CXX object libnestutil/CMakeFiles/nestutil.dir/stopwatch.cpp.o
[  1%] Linking CXX static library libnestutil.a
[  1%] Built target nestutil
[  1%] Building CXX object sli/CMakeFiles/sli_lib.dir/allocator.cpp.o
[  2%] Building CXX object sli/CMakeFiles/sli_lib.dir/arraydatum.cc.o
[  2%] Building CXX object sli/CMakeFiles/sli_lib.dir/booldatum.cc.o
[  2%] Building CXX object sli/CMakeFiles/sli_lib.dir/charcode.cc.o
[  3%] Building CXX object sli/CMakeFiles/sli_lib.dir/datum.cc.o
[  3%] Building CXX object sli/CMakeFiles/sli_lib.dir/dict.cc.o
[  4%] Building CXX object sli/CMakeFiles/sli_lib.dir/dictstack.cc.o
[  4%] Building CXX object sli/CMakeFiles/sli_lib.dir/dictutils.cc.o
[  4%] Building CXX object sli/CMakeFiles/sli_lib.dir/doubledatum.cc.o
[  5%] Building CXX object sli/CMakeFiles/sli_lib.dir/fdstream.cc.o
[  5%] Building CXX object sli/CMakeFiles/sli_lib.dir/filesystem.cc.o
[  5%] Building CXX object sli/CMakeFiles/sli_lib.dir/functiondatum.cc.o
[  6%] Building C object sli/CMakeFiles/sli_lib.dir/get_mem.c.o
[  6%] Building CXX object sli/CMakeFiles/sli_lib.dir/integerdatum.cc.o
[  7%] Building CXX object sli/CMakeFiles/sli_lib.dir/interpret.cc.o
[  7%] Building CXX object sli/CMakeFiles/sli_lib.dir/literaldatum.cc.o
[  7%] Building CXX object sli/CMakeFiles/sli_lib.dir/name.cc.o
[  8%] Building CXX object sli/CMakeFiles/sli_lib.dir/slinames.cpp.o
[  8%] Building CXX object sli/CMakeFiles/sli_lib.dir/namedatum.cc.o
[  8%] Building CXX object sli/CMakeFiles/sli_lib.dir/oosupport.cc.o
[  9%] Building CXX object sli/CMakeFiles/sli_lib.dir/parser.cc.o
[  9%] Building CXX object sli/CMakeFiles/sli_lib.dir/processes.cc.o
[  9%] Building CXX object sli/CMakeFiles/sli_lib.dir/scanner.cc.o
[ 10%] Building CXX object sli/CMakeFiles/sli_lib.dir/sli_io.cc.o
[ 10%] Building CXX object sli/CMakeFiles/sli_lib.dir/sliactions.cc.o
[ 11%] Building CXX object sli/CMakeFiles/sli_lib.dir/sliarray.cc.o
[ 11%] Building CXX object sli/CMakeFiles/sli_lib.dir/slibuiltins.cc.o
[ 11%] Building CXX object sli/CMakeFiles/sli_lib.dir/slicontrol.cc.o
[ 12%] Building CXX object sli/CMakeFiles/sli_lib.dir/slidata.cc.o
[ 12%] Building CXX object sli/CMakeFiles/sli_lib.dir/slidict.cc.o
[ 12%] Building CXX object sli/CMakeFiles/sli_lib.dir/sliexceptions.cc.o
[ 13%] Building CXX object sli/CMakeFiles/sli_lib.dir/sligraphics.cc.o
[ 13%] Building CXX object sli/CMakeFiles/sli_lib.dir/slimath.cc.o
[ 14%] Building CXX object sli/CMakeFiles/sli_lib.dir/slimodule.cc.o
[ 14%] Building CXX object sli/CMakeFiles/sli_lib.dir/sliregexp.cc.o
[ 14%] Building CXX object sli/CMakeFiles/sli_lib.dir/slistack.cc.o
[ 15%] Building CXX object sli/CMakeFiles/sli_lib.dir/slistartup.cc.o
[ 15%] Building CXX object sli/CMakeFiles/sli_lib.dir/slitype.cc.o
[ 15%] Building CXX object sli/CMakeFiles/sli_lib.dir/slitypecheck.cc.o
[ 16%] Building CXX object sli/CMakeFiles/sli_lib.dir/specialfunctionsmodule.cc.o
[ 16%] Building CXX object sli/CMakeFiles/sli_lib.dir/stringdatum.cc.o
[ 16%] Building CXX object sli/CMakeFiles/sli_lib.dir/symboldatum.cc.o
[ 17%] Building CXX object sli/CMakeFiles/sli_lib.dir/tarrayobj.cc.o
[ 17%] Building CXX object sli/CMakeFiles/sli_lib.dir/token.cc.o
[ 18%] Building CXX object sli/CMakeFiles/sli_lib.dir/tokenarray.cc.o
[ 18%] Building CXX object sli/CMakeFiles/sli_lib.dir/tokenstack.cc.o
[ 18%] Building CXX object sli/CMakeFiles/sli_lib.dir/tokenutils.cc.o
[ 19%] Building CXX object sli/CMakeFiles/sli_lib.dir/triedatum.cc.o
[ 19%] Building CXX object sli/CMakeFiles/sli_lib.dir/typechk.cc.o
[ 19%] Building CXX object sli/CMakeFiles/sli_lib.dir/utils.cc.o
[ 20%] Linking CXX shared library libsli.so
[ 20%] Built target sli_lib
[ 20%] Building CXX object sli/CMakeFiles/sli_readline.dir/gnureadline.cc.o
[ 20%] Building CXX object nestkernel/CMakeFiles/nestkernel.dir/archiving_node.cpp.o
[ 20%] Building CXX object nestkernel/CMakeFiles/nestkernel.dir/clopath_archiving_node.cpp.o
[ 21%] Building CXX object nestkernel/CMakeFiles/nestkernel.dir/eprop_archiving_node.cpp.o
[ 21%] Building CXX object nestkernel/CMakeFiles/nestkernel.dir/common_synapse_properties.cpp.o
[ 21%] Building CXX object nestkernel/CMakeFiles/nestkernel.dir/connector_model.cpp.o
[ 22%] Building CXX object nestkernel/CMakeFiles/nestkernel.dir/deprecation_warning.cpp.o
[ 22%] Building CXX object nestkernel/CMakeFiles/nestkernel.dir/connection_id.cpp.o
[ 23%] Linking CXX shared library libsli_readline.so
In file included from /mnt/c/Users/pwd06/git/nest-simulator/libnestutil/sort.h:36,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/connector_base.h:35,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/connection_manager.h:36,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/kernel_manager.h:30,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/deprecation_warning.cpp:26:
/mnt/c/Users/pwd06/git/nest-simulator/libnestutil/iterator_pair.h:26:10: fatal error: boost/iterator/iterator_adaptor.hpp: No such file or directory
   26 | #include <boost/iterator/iterator_adaptor.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [nestkernel/CMakeFiles/nestkernel.dir/build.make:160: nestkernel/CMakeFiles/nestkernel.dir/deprecation_warning.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /mnt/c/Users/pwd06/git/nest-simulator/libnestutil/sort.h:36,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/connector_base.h:35,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/connection_manager.h:36,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/kernel_manager.h:30,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/archiving_node.cpp:26:
/mnt/c/Users/pwd06/git/nest-simulator/libnestutil/iterator_pair.h:26:10: fatal error: boost/iterator/iterator_adaptor.hpp: No such file or directory
   26 | #include <boost/iterator/iterator_adaptor.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [nestkernel/CMakeFiles/nestkernel.dir/build.make:76: nestkernel/CMakeFiles/nestkernel.dir/archiving_node.cpp.o] Error 1
In file included from /mnt/c/Users/pwd06/git/nest-simulator/libnestutil/sort.h:36,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/connector_base.h:35,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/connection_manager.h:36,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/kernel_manager.h:30,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/eprop_archiving_node_impl.h:29,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/eprop_archiving_node.cpp:25:
/mnt/c/Users/pwd06/git/nest-simulator/libnestutil/iterator_pair.h:26:10: fatal error: boost/iterator/iterator_adaptor.hpp: No such file or directory
   26 | #include <boost/iterator/iterator_adaptor.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [nestkernel/CMakeFiles/nestkernel.dir/build.make:104: nestkernel/CMakeFiles/nestkernel.dir/eprop_archiving_node.cpp.o] Error 1
In file included from /mnt/c/Users/pwd06/git/nest-simulator/libnestutil/sort.h:36,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/connector_base.h:35,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/connection_manager.h:36,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/kernel_manager.h:30,
                 from /mnt/c/Users/pwd06/git/nest-simulator/models/weight_recorder.h:33,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/common_synapse_properties.cpp:32:
/mnt/c/Users/pwd06/git/nest-simulator/libnestutil/iterator_pair.h:26:10: fatal error: boost/iterator/iterator_adaptor.hpp: No such file or directory
   26 | #include <boost/iterator/iterator_adaptor.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [nestkernel/CMakeFiles/nestkernel.dir/build.make:118: nestkernel/CMakeFiles/nestkernel.dir/common_synapse_properties.cpp.o] Error 1
[ 23%] Built target sli_readline
In file included from /mnt/c/Users/pwd06/git/nest-simulator/libnestutil/sort.h:36,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/connector_base.h:35,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/connection_manager.h:36,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/kernel_manager.h:30,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/clopath_archiving_node.cpp:26:
/mnt/c/Users/pwd06/git/nest-simulator/libnestutil/iterator_pair.h:26:10: fatal error: boost/iterator/iterator_adaptor.hpp: No such file or directory
   26 | #include <boost/iterator/iterator_adaptor.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [nestkernel/CMakeFiles/nestkernel.dir/build.make:90: nestkernel/CMakeFiles/nestkernel.dir/clopath_archiving_node.cpp.o] Error 1
[ 24%] Building CXX object sli/CMakeFiles/sli.dir/puresli.cc.o
In file included from /usr/include/c++/10/cassert:44,
                 from /mnt/c/Users/pwd06/git/nest-simulator/libnestutil/lockptr.h:31,
                 from /mnt/c/Users/pwd06/git/nest-simulator/sli/lockptrdatum.h:27,
                 from /mnt/c/Users/pwd06/git/nest-simulator/sli/dictdatum.h:32,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/manager_interface.h:30,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/vp_manager.h:27,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/event.h:37,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/connector_model.h:35,
                 from /mnt/c/Users/pwd06/git/nest-simulator/nestkernel/connector_model.cpp:23:
/mnt/c/Users/pwd06/git/nest-simulator/nestkernel/vp_manager.h: In member function ‘void nest::VPManager::assert_thread_parallel() const’:
/mnt/c/Users/pwd06/git/nest-simulator/nestkernel/vp_manager.h:206:33: warning: comparison of integer expressions of different signedness: ‘int’ and ‘const size_t’ {aka ‘const long unsigned int’} [-Wsign-compare]
  206 |   assert( omp_get_num_threads() == n_threads_ );
      |           ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
make[1]: *** [CMakeFiles/Makefile2:715: nestkernel/CMakeFiles/nestkernel.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 24%] Linking CXX executable sli
[ 24%] Built target sli
make: *** [Makefile:146: all] Error 2
```

**Comment by Helveg** [comment]
The problem was that my `BOOST_ROOT` wasn't set to my installation, and some other Boost was picked up, without any libraries. The underlying problem is that `NEST_PROCESS_WITH_BOOST` uses `find_package( Boost 1.69.0 )`, but does not add which libraries are required with for example `find_package( Boost 1.69.0 COMPONENTS date_time filesystem iostreams)`.

**Comment by Helveg** [comment]
When trying to submit a PR to address this I found out that CMake's `FindBoost` does not add components for header only libraries. Since NEST only uses `Boost.Iterator`, `Boost.Sort`, `Boost.Test`, and `Boost.Tuple', and they are all header only, there does not seem to be a great way to catch this error earlier on.

Closing this and hoping that the error log is sufficient for others to find this if they have a broken Boost installation somewhere.
